lib/utils/datasetManager.hpp

```diff
@@ -500,16 +539,41 @@ class datasetManager {
 template<typename T>
 inline const T* datasetManager::dataset_state(dset_id_t dset) {
 
-    if (!_use_broker)
-        return get_closest_ancestor<T>(dset);
+    // Try to find a matching dataset
+    std::string type = FACTORY(datasetState)::label<T>();
+    auto ret = closest_dataset_of_type(dset, type);
+
+    // If not found, return null
+    if (!ret)
+        return nullptr;
+
+    state_id_t state_id = ret.value().second.state();
 
-    // get an update on the dataset topology (blocking)
-    update_datasets(dset);
+    {
+        std::lock_guard<std::mutex> dslock(_lock_dsets);
```

Comment:
Consequences of releasing the lock between updating the datasets and figuring out which states to request could mean that states are requested multiple times. May not be a big problem. but should keep it in mind?

Reply:
Good point. I'll put a comment in the source.